### PR TITLE
Un-orphan instances

### PR DESCRIPTION
**Pull request by phadej**
Resolves https://github.com/bos/aeson/issues/375

### CODE REVIEW

**Comment by phadej** [comment]
@andrewthad @bergmark this is ready for review

File: Data/Aeson/Types/Generic.hs [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
+{-# LANGUAGE DataKinds              #-}
+{-# LANGUAGE EmptyDataDecls         #-}
+{-# LANGUAGE FlexibleInstances      #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE KindSignatures         #-}
+{-# LANGUAGE MultiParamTypeClasses  #-}
+{-# LANGUAGE ScopedTypeVariables    #-}
+{-# LANGUAGE TypeOperators          #-}
+{-# LANGUAGE UndecidableInstances   #-}
 {-# OPTIONS_GHC -fno-warn-orphans #-}
```
Comment (andrewthad) [inline Data/Aeson/Types/Generic.hs]: We can get rid of `-fno-warn-orphans`.

Comment (phadej) [inline Data/Aeson/Types/Generic.hs]: Done

**Comment by andrewthad** [comment]
I've given it a cursory glance, and this looks how I would expect it to. I'm +1 on merging into master. There's one place where you left an `-fno-warn-orphans` flag on that I think can be removed though.

**Pull request merged**